Update genai-guide-npu.rst (#26612)
openvinotoolkit/openvino: docs/articles_en/learn-openvino/llm_inference_guide/genai-guide-npu.rst

   optimum-cli export openvino -m TinyLlama/TinyLlama-1.1B-Chat-v1.0 --weight-format int4 --sym --group-size 128 --ratio 1.0 TinyLlama

> Note: For models exceeding 1 billion parameters, it's recommended to use channel-wise quantization. This approach is particularly effective, as demonstrated in the example with the llama-2-7b-chat-hf model:
> .. code-block:: python
> 
>    optimum-cli export openvino -m meta-llama/Llama-2-7b-chat-hf --weight-format int4 --sym --group-size -1 --ratio 1.0 Llama-2-7b-chat-hf

Reviewer: ```suggestion
**For models exceeding 1 billion parameters, it is recommended to use remarkably effective channel-wise quantization.** For example, you can try the approach with the llama-2-7b-chat-hf model:

.. code-block:: python

   optimum-cli export openvino -m meta-llama/Llama-2-7b-chat-hf --weight-format int4 --sym --group-size -1 --ratio 1.0 Llama-2-7b-chat-hf
```

   optimum-cli export openvino -m TinyLlama/TinyLlama-1.1B-Chat-v1.0 --weight-format int4 --sym --group-size 128 --ratio 1.0 TinyLlama

**For models exceeding 1 billion parameters, it is recommended to use remarkably effective channel-wise quantization.** For example, you can try the approach with the llama-2-7b-chat-hf model:

.. code-block:: python

   optimum-cli export openvino -m meta-llama/Llama-2-7b-chat-hf --weight-format int4 --sym --group-size -1 --ratio 1.0 Llama-2-7b-chat-hf

Run generation using OpenVINO GenAI
###################################

> It is recommended to install the latest available driver: https://www.intel.com/content/www/us/en/download/794734/intel-npu-driver-windows.html

Reviewer: ```suggestion
It is recommended to install the latest available `driver <https://www.intel.com/content/www/us/en/download/794734/intel-npu-driver-windows.html>`__.
```

You may configure both the 'maximum input prompt length' and 'minimum response length' using the following parameters:

- ``MAX_PROMPT_LEN``: Defines the maximum number of tokens that the LLM pipeline can process for the input prompt (default: 1024).
- ``MIN_RESPONSE_LEN``: Specifies the minimum number of tokens that the LLM pipeline will generate in its response (default: 150).

Use the following code snippet to change the default settings:

.. tab-set::

   .. tab-item:: Python
      :sync: py

      .. code-block:: python

         pipeline_config = { "MAX_PROMPT_LEN": 1500, "MIN_RESPONSE_LEN": 500 }
         pipe = ov_genai.LLMPipeline(model_path, "NPU", pipeline_config)

   .. tab-item:: C++
      :sync: cpp

      .. code-block:: cpp

         ov::AnyMap pipeline_config = { { "MAX_PROMPT_LEN",  1500 }, { "MIN_RESPONSE_LEN", 500 } };
         ov::genai::LLMPipeline pipe(model_path, "NPU", pipeline_config);

Compilation options
+++++++++++++++++++

> Compiling models for NPU may take a while. By default, the LLMPipeline for the NPU

Author: There was also concern if we should still recommend such options (CC: @dmatveev)

Author: Removed

Additional configuration options
################################

Prompt and response length options
##################################

The LLM pipeline for NPUs leverages a static shape approach, optimizing performance during execution. However, this method may introduce certain usage limitations. By default, the LLM pipeline supports input prompts up to 1024 tokens in length and ensures that the generated response contains at least 150 tokens, unless the generation encounters an end-of-sequence (EOS) token or the user explicitly sets a lower limit for the response length.

Both the maximum input prompt length and minimum response length are configurable using the parameters ``MAX_PROMPT_LEN`` and ``MIN_RESPONSE_LEN``:
- **``MAX_PROMPT_LEN``**: Defines the maximum number of tokens that the LLM pipeline can process for the input prompt (default: 1024).
- **``MIN_RESPONSE_LEN``**: Specifies the minimum number of tokens that the LLM pipeline will generate in its response (default: 150).

> Use the following code snippet to extend default lenghts settings:

Reviewer 2: lenghts -> lengths

Author: Thank you - done!

Additional configuration options
################################

> Prompt and response length options

Author: Can it be smaller than "Additional configuration options"?

##################################

> The LLM pipeline for NPUs leverages a static shape approach, optimizing performance during execution. However, this method may introduce certain usage limitations. By default, the LLM pipeline supports input prompts up to 1024 tokens in length and ensures that the generated response contains at least 150 tokens, unless the generation encounters an end-of-sequence (EOS) token or the user explicitly sets a lower limit for the response length.

Reviewer: ```suggestion
The LLM pipeline for NPUs leverages the static shape approach by optimizing performance. However, it may introduce certain usage limitations. By default, the LLM pipeline supports input prompts up to 1024 tokens in length. It also ensures that the generated response contains at least 150 tokens, unless the generation encounters an end-of-sequence (EOS) token or the user explicitly sets a lower limit for the response length.
```

Reviewer 3: I'd be cautious here: ", optimizing" is not the same as "by optimizing" My proposition would be:

```suggestion
The LLM pipeline for NPUs leverages the static shape approach, optimizing execution performance, while potentially introducing certain usage limitations. By default, the LLM pipeline supports input prompts up to 1024 tokens in length. It also ensures that the generated response contains at least 150 tokens, unless the generation encounters the end-of-sequence (EOS) token or the user explicitly sets a lower length limit for the response.
```

> Both the maximum input prompt length and minimum response length are configurable using the parameters ``MAX_PROMPT_LEN`` and ``MIN_RESPONSE_LEN``:

Reviewer: ```suggestion
Both the maximum length of input prompt and minimum length of response are configurable with the ``MAX_PROMPT_LEN`` and ``MIN_RESPONSE_LEN`` parameters:
```

Reviewer 3: ```suggestion
You may configure both the 'maximum input prompt length' and 'minimum response length' using the following parameters:
```

> - **``MAX_PROMPT_LEN``**: Defines the maximum number of tokens that the LLM pipeline can process for the input prompt (default: 1024).

Reviewer: ```suggestion

- **``MAX_PROMPT_LEN``**: Defines the maximum number of tokens that the LLM pipeline can process for the input prompt (default: 1024).
```

> - **``MIN_RESPONSE_LEN``**: Specifies the minimum number of tokens that the LLM pipeline will generate in its response (default: 150).

Reviewer: ```suggestion
- ``MIN_RESPONSE_LEN``: Specifies the minimum number of tokens that the LLM pipeline will generate in its response (default: 150).
```

> Use the following code snippet to extend default lengths settings:

Reviewer: ```suggestion
Use the following code snippet to change the default settings:
```

.. tab-set::

   .. tab-item:: Python
      :sync: py

      .. code-block:: python

         pipeline_config = { "MAX_PROMPT_LEN": 1500, "MIN_RESPONSE_LEN": 500 }
         pipe = ov_genai.LLMPipeline(model_path, "NPU", pipeline_config)

   .. tab-item:: C++
      :sync: cpp

      .. code-block:: cpp

         ov::AnyMap pipeline_config = { { "MAX_PROMPT_LEN",  1500 }, { "MIN_RESPONSE_LEN", 500 } };
         ov::genai::LLMPipeline pipe(model_path, "NPU", pipeline_config);

Compilation options

> ###################

Reviewer 3: ```suggestion
+++++++++++++++++++
```
